catalog/imaging/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java

```diff
@@ -0,0 +1,667 @@
+
+            handleSegment(parsingStrategy.getTextSegmentHeaders(),
+                    (text) -> setMetadataForTextHeader(metadataXml, text));
+
+            metadataXml.append("</metadata>\n");
+            metacard.setAttribute(Metacard.METADATA, metadataXml.toString());
+
+            // Set GEOGRAPHY from discovered polygons
+            if (polygonList.size() == 1) {
+                metacard.setAttribute(Metacard.GEOGRAPHY,
+                        polygonList.get(0)
+                                .toText());
+            } else if (polygonList.size() > 1) {
+                Polygon[] polyAry = polygonList.toArray(new Polygon[0]);
+                MultiPolygon multiPolygon = GEOMETRY_FACTORY.createMultiPolygon(polyAry);
+                metacard.setAttribute(Metacard.GEOGRAPHY, multiPolygon.toText());
+            }
+
+
+            //TODO: Generate thumbnails - this doesn't work in a headless environment
+/*
+            ImageInputStream imageData = new MemoryCacheImageInputStream(new ByteArrayInputStream(parsingStrategy.getImageSegmentData().get(0)));
+            NitfRenderer renderer = new NitfRenderer();
+            BufferedImage img = renderer.render(parsingStrategy.getImageSegmentHeaders().get(0), imageData);
+            img = scale(img);
+
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ImageIO.write(img, "jpg", bos);
+            if (bos.size() > 1024*128) {
+                LOGGER.info("Thumbnail too big, ignoring.");
+            } else {
+                metacard.setAttribute(Metacard.THUMBNAIL, bos.toByteArray());
+            }
+*/
+
+
+        } catch (ParseException e) {
+            throw new CatalogTransformerException(e);
+        }
+    }
+
+    private <T extends CommonNitfSegment> void handleSegment(List<T> segmentList, Consumer<T> segmentConsumer) {
+        for (T segment : segmentList) {
+            segmentConsumer.accept(segment);
+        }
+    }
+    private void setHeaderAttributes(DynamicMetacard metacard, NitfFileHeader header) {
+        Date date = getDate(header.getFileDateTime()
+                .getSourceString());
+        metacard.setAttribute(Metacard.CREATED, date);
+        metacard.setAttribute(Metacard.TITLE, header.getFileTitle());
+        metacard.setAttribute(Nitf.NITF_VERSION, header.getFileType());
+        metacard.addAttribute(Nitf.FILE_DATE_TIME, date);
+        // TODO: modified date from HISTOA?
+        metacard.setAttribute(Metacard.MODIFIED, date);
+        metacard.setAttribute(Metacard.CREATED, date);
+        metacard.setAttribute(Metacard.EFFECTIVE, date);
+
+        metacard.addAttribute(Nitf.FILE_TITLE, header.getFileTitle());
+        metacard.addAttribute(Nitf.COMPLEXITY_LEVEL, header.getComplexityLevel());
+        metacard.addAttribute(Nitf.ORIGINATOR_NAME, header.getOriginatorsName());
+        metacard.addAttribute(Nitf.ORIGINATING_STATION_ID, header.getOriginatingStationId());
+
+        metacard.addAttribute(Nitf.CODE_WORDS,
```

Comment:
Might want to put the object returned by `getFileSecurityMetadata()` into a local variable.